dexec/cmd.go

```diff
@@ -120,35 +120,44 @@ func (c *Cmd) logiofn(prefix string) func(string) {
 		if c.Process != nil {
 			pid = c.Process.Pid
 		}
-		dlog.Printf(c.ctx, "[pid:%v] %s %s", pid, prefix, msg)
+		ctx := dlog.WithField(c.ctx, "dexec.pid", pid)
+		ctx = dlog.WithField(ctx, "dexec.stream", stream)
+		if msg != nil {
+			ctx = dlog.WithField(ctx, "dexec.data", string(msg))
+		}
+		if err != nil {
+			ctx = dlog.WithField(ctx, "dexec.err", err)
+		}
+		dlog.Print(ctx)
 	}
 }
 
 // Start starts the specified command but does not wait for it to complete.
 //
 // See the os/exec.Cmd.Start documenaton for more information.
 func (c *Cmd) Start() error {
-	c.Stdin = fixupReader(c.Stdin, c.logiofn("stdin  <"))
+	c.Stdin = fixupReader(c.Stdin, c.logiofn("stdin"))
 	if interfaceEqual(c.Stdout, c.Stderr) {
-		c.Stdout = fixupWriter(c.Stdout, c.logiofn("stdout+stderr >"))
+		c.Stdout = fixupWriter(c.Stdout, c.logiofn("stdout+stderr"))
 		c.Stderr = c.Stdout
 	} else {
-		c.Stdout = fixupWriter(c.Stdout, c.logiofn("stdout >"))
-		c.Stderr = fixupWriter(c.Stderr, c.logiofn("stderr >"))
+		c.Stdout = fixupWriter(c.Stdout, c.logiofn("stdout"))
+		c.Stderr = fixupWriter(c.Stderr, c.logiofn("stderr"))
 	}
 
 	err := c.Cmd.Start()
 	if err == nil {
 		if !c.DisableLogging {
-			dlog.Printf(c.ctx, "[pid:%v] started command %#v", c.Process.Pid, c.Args)
+			ctx := dlog.WithField(c.ctx, "dexec.pid", c.Process.Pid)
+			dlog.Printf(ctx, "started command %q", c.Args)
```

Comment:
You might want to consider doing something similar to what Telepresence's `ShellString()` does instead of just using `%q`. Those `\"`s are not friendly to the eye: https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/client/logging/shellstring.go

Reply:
I tried shell-quoting it, and while there is certainly still room for improvement, I found this version to be more readable than the shell-quoted version.